api/v1beta2/condition_types.go

```diff
@@ -100,4 +100,7 @@ const (
 
 	// CacheOperationFailedReason signals a failure in cache operation.
 	CacheOperationFailedReason string = "CacheOperationFailed"
+
+	// ControllerUndefinedBehaviorReason signals an undefined behavior by the controller.
+	ControllerUndefinedBehaviorReason string = "ControllerUndefinedBehavior"
 )
```

Comment:
I find these terms very machine-like, to a user, something like "unsupported feature" would probably be a lot more self explanatory.

Reply:
How about `ControllerMisbehaviorReason`?

Reply:
How is it misbehaving? It is simply running into a limitation due to something not being enabled.

Reply:
Yeah, but "unsupported feature" doesn't seem accurate as well, as the feature _is_ supported, but it is not being able to fulfill it.

Reply:
Yes, so I wouldn't make the reason part of the API (because they officially are arbitrary), and simply state something like `TransportNotEnabled`.